### ISSUE TITLE
Question about replace_pattern

### ISSUE THREAD

**Issue by IvanYashchuk** [ELS-RD/kernl#180]
Kernl has its own `replace_pattern` function in `kernl.utils.extended_matcher`. What are the reasons to use that instead of `torch.fx.replace_pattern`? Is there anything you would like to see changed or fixed in `torch.fx.replace_pattern`?

**Comment by gaetansnl** [comment]
Hello,
We started using it at the time the replace_pattern of torch contained bugs and did not support comparaison of args, so we copied it and manually corrected bugs to fit our use case. Since then, pytorch fixed many bugs and we updated some parts to use the latest code from pytorch, but we did not check recently the state of replace_pattern in pytorch repository.

The main change we have is that when we replace a pattern containing a module (for example linear), instead of having a new linear in the replacement pattern, it is replaced by the original linear, so we can access weight in the replacement pattern.

An awesome feature would be to be able to match any constant in the original pattern, and access the constant as argument in the replacement pattern.
